### DISCUSSION TITLE
Energy in MML and AT2

### DISCUSSION THREAD

**Discussion by oscarxblanco** [atcollab/at#671] [Q&A]
Dear all,
when getting brho from MML using 'getbrho.m' the electron rest energy is added up to the `energy` parameter, see line 57 :
https://github.com/atcollab/MML/blob/5d622d6ffc8689e5e1d015069d29c302734bf9b0/mml/getbrho.m#L57C34-L57C34
This would mean that the parameter `energy` is not the total beam energy, but, something else.

As for AT2, I understand the ring `energy` is the total energy. For example in atsummary.m :
https://github.com/atcollab/at/blob/1a0707f7c347db20b9b543e67cd1a2f8c5c53417/atmat/atphysics/ParameterSummaryFunctions/atsummary.m#L63
https://github.com/atcollab/at/blob/1a0707f7c347db20b9b543e67cd1a2f8c5c53417/atmat/atphysics/ParameterSummaryFunctions/atsummary.m#L86C43-L86C43

Does it mean that AT2 and MML use different definitions of energy ?

**Comment by oscarxblanco** [comment]
I have reported a possible error in MML here 
https://github.com/atcollab/MML/issues/6

Let me know your opinion.
o

**Reply by lnadolski** [reply]
dear All;

For me in AT, the energy in AT is kinetic energy and not the full energy.
This is relevant, especially for low energy like the LINAC to booster transfer lines.

**Comment by lfarv** [comment]
Hi all,

In AT (I cannot talk for MML), the `energy` is the **total energy**: $E=\gamma E_0$. This is consistent all around the Matlab and python codes, including the C integrators. Changing this convention would be extremely dangerous. But we can easily add a new lattice property (`kinetic_energy`?) for convenience.

**Comment by lfarv** [comment]
May I also remind that the tracking in AT works only for relativistic particles ( $\beta == 1$ )? The `particle` attribute was introduced to allow non-relativistic, and there is a branch for [non-relativistic tracking](https://github.com/atcollab/at/tree/non-relativistic_tracking), but I have not been working on it for long…

**Comment by oscarxblanco** [comment]
Dear @lfarv  and @lnadolski ,
thank you for the answer. So, what I understand is that:
-  the error is in MML, 
- and AT only works on ultrarelativistic approach where $1-\beta$ is very small to be relevant, in practice on the order of $10^{-3}$ or even $10^{-2}$ if 1% error is not important.

I waited a few weeks but it seems there are no other comments. If you prefer this question could be closed.

Best regards,
o
